core/src/main/java/org/elasticsearch/index/query/BaseTermQueryBuilder.java

```diff
@@ -146,18 +146,18 @@ public QueryValidationException validate() {
     }
 
     @Override
-    protected final int doHashCode() {
-        return Objects.hash(getClass(), fieldName, value);
+    protected int doHashCode() {
+        return Objects.hash(fieldName, value);
     }
 
     @Override
-    protected final boolean doEquals(BaseTermQueryBuilder other) {
+    protected boolean doEquals(BaseTermQueryBuilder other) {
         return Objects.equals(fieldName, other.fieldName) &&
                Objects.equals(value, other.value);
     }
 
     @Override
-    protected final QB doReadFrom(StreamInput in) throws IOException {
```

Comment:
I think we should allow these to be overridden, but I'm happy to make the change.